src/libcore/slice/mod.rs

```diff
@@ -1459,6 +1459,68 @@ impl<T> [T] {
         m >= n && needle == &self[m - n..]
     }
 
+    /// Returns a subslice with the prefix removed.
+    ///
+    /// This method returns the original slice if `prefix` is an empty slice
+    /// or returns [`None`] if slice does not start with `prefix`.
```

Comment:
I think the wording of this was clearer in a previous revision. Slice not starting with `prefix` is a scenario that is important to specify; this says what the meaning of Option in the return type is. Empty `prefix` is a small clarification (not even an edge case, since the behavior falls out from the rest of the documentation). I think the emphasis in this paragraph is backward and not conducive to recognizing the part that is more important.

Reply:
I repharsed it. Could you take a look?